src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,145 @@
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.artifacts.PublishArtifact
+import org.gradle.api.tasks.TaskAction
+import org.gradle.plugins.signing.Signature
+
+class OreDeployTask extends DefaultTask {
+
+    @TaskAction
+    void deploy() {
+        def artifacts = project.configurations.archives.allArtifacts
+        PublishArtifact plugin = artifacts.find { a -> (a.type == "jar") }
+        PublishArtifact sig = artifacts.find { a -> (a instanceof Signature) }
+        if (plugin == null) {
+            throw new GradleException("Plugin file not found.")
+        }
+        if (sig == null) {
+            throw new GradleException("Signature file not found.")
+        }
+
+        String instanceUrl = project.oreDeploy.instanceUrl
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+        } catch (MalformedURLException e) {
+            throw new GradleException("Invalid project URL", e)
+        }
+
+        String apiKey = project.oreDeploy.apiKey
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        String channel = project.oreDeploy.channel
+        if (channel == null) {
+            throw new GradleException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
+                .addPart("apiKey", new StringBody(apiKey))
+                .addPart("channel", new StringBody(channel))
+                .addPart("pluginFile", new FileBody(plugin.file))
+                .addPart("pluginSig", new FileBody(sig.file))
+                .build()
+
+        HttpURLConnection conn = null
+        OutputStream os = null
+        InputStream is = null
+        try {
+            conn = (HttpURLConnection) projectUrl.openConnection()
+            conn.doOutput = true
+            conn.requestMethod = 'POST'
+            conn.useCaches = false
+            conn.readTimeout = 10000
+            conn.connectTimeout = 10000
+            conn.setRequestProperty('Connection', 'Keep-Alive')
+            conn.setRequestProperty('Cache-Control', 'no-cache')
+            conn.setRequestProperty('Content-Length', requestEntity.contentLength as String)
+            conn.setRequestProperty('Content-Type', requestEntity.contentType.value)
+
+            os = conn.outputStream
+            requestEntity.writeTo(os)
+            os.flush()
+            os.close()
+            os = null
+
+            conn.connect()
+            if (conn.responseCode != HttpURLConnection.HTTP_CREATED) {
+                println '[failure] ' + conn.responseCode + ' ' + conn.responseMessage
```

Comment:
You should use the logger of the task here (just get the `logger` property of the task)